### DISCUSSION TITLE
Dynamic Dialog - maximise dialog with custom header

### DISCUSSION THREAD

**Discussion by nilsm0hr** [primefaces/community#1401] [PrimeNG] [answered]
Hi, 

I have created a custom header for a dynamic dialog and would like to add a maximise button to the header. The close button works just fine but the maximise function does not work as expected.

Am I doing something wrong here? I wasn't able to find much in the docs on how to achieve this. 

I also set maximizable = ture. 

`maximizeDialog() {`
`    this.maximized = !this.maximized;`
`    this.ref.maximize({maximized: this.maximized});`
`}`
`  closeDialog() {`
`    this.ref.close();`
`}`

Thanks

**Comment by cetincakiroglu** [comment]
Hi,

Could you please share a stackblitz example so we can identify the issue?

**Comment by nilsm0hr** [comment]
Hi, 
here is the stackblitz. I have created a custom header cause I want the two chevron buttons for better navigation in the header plus some other bits of information. 

The close, cancel, and selection actions work great but for maximize, I just see a toast with maximize = true/false but nothing happens.

https://stackblitz.com/edit/bncwab

**Reply by cetincakiroglu** [reply]
Hi,

Thanks a lot for the stackblitz example, I've added it to the 17.13.0 milestone. We'll try to address this issue in next week's release.
 issue link: https://github.com/primefaces/primeng/issues/15113

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

We've reviewed the issue and recently we've added `getInstance` method to dialogService to make it easier to access and use methods of dynamic dialog. You can use instance to access and trigger maximize in wherever you want. Here is the refactored version of your example: https://stackblitz.com/edit/bncwab-eitauj?file=src%2Fapp%2Fdemo%2Ffooter.ts,src%2Fapp%2Fdemo%2Fheader.ts

**Reply by nilsm0hr** [reply]
Hi, thanks so much. works perfectly now
